src/ucp/rndv/rndv_am.c

```diff
@@ -135,3 +135,96 @@ ucp_proto_t ucp_rndv_am_bcopy_proto = {
     .abort    = ucp_proto_rndv_am_bcopy_abort,
     .reset    = ucp_proto_reset_fatal_not_implemented
 };
+
+static UCS_F_ALWAYS_INLINE ucs_status_t ucp_rndv_am_zcopy_send_func(
+        ucp_request_t *req, const ucp_proto_multi_lane_priv_t *lpriv,
+        ucp_datatype_iter_t *next_iter, ucp_lane_index_t *lane_shift)
+{
+    const size_t hdr_size      = sizeof(ucp_request_data_hdr_t);
+    const size_t max_payload   = ucp_proto_multi_max_payload(req, lpriv,
+                                                             hdr_size);
+    ucp_request_data_hdr_t hdr = {
+        .req_id = req->send.rndv.remote_req_id,
+        .offset = req->send.state.dt_iter.offset
+    };
+    uct_iov_t iov[UCP_MAX_IOV];
+    size_t iov_count;
+
+    ucs_assert(lpriv->super.max_iov > 1);
+    iov_count = ucp_datatype_iter_next_iov(&req->send.state.dt_iter,
+                                           max_payload, lpriv->super.md_index,
+                                           UCP_DT_MASK_CONTIG_IOV, next_iter,
+                                           iov, lpriv->super.max_iov - 1);
```

Comment:
We do not need to reserve iov here.
`lpriv->super.max_iov - 1` was done in "am/egr/multi/zcopy" to reserve space for footer.

The assert at line 153 also should be changed: 1 -> 0